sktime/annotation/base/_base.py

```diff
@@ -0,0 +1,256 @@
+
+        self._X = None
+        self._Y = None
+
+        super(BaseSeriesAnnotator, self).__init__()
+
+    def fit(self, X, Y=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            training data to fit model to, time series
+        Y : pd.Series, optional
+            ground truth annotations for training if annotator is supervised
+        Returns
+        -------
+        self : returns a reference to self
+
+        State change
+        ------------
+        creates fitted model (attributes ending in "_")
+        sets _is_fitted flag to true
+        """
+
+        X = check_series(X)
+
+        if Y is not None:
+            Y = check_series(Y)
+
+        self._X = X
+        self._Y = Y
+
+        # fkiraly: insert checks/conversions here, after PR #1012 I suggest
+
+        self._fit(X=X, Y=Y)
+
+        # this should happen last
+        self._is_fitted = True
+
+        return self
+
+    def predict(self, X):
+        """Create annotations on test/deployment data.
+
+        Parameters
+        ----------
+        X : pd.DataFrame - data to annotate, time series
+
+        Returns
+        -------
+        Y : pd.Series - annotations for sequence X
+            exact format depends on annotation type
+        """
+
+        self.check_is_fitted()
+
+        X = check_series(X)
+
+        # fkiraly: insert checks/conversions here, after PR #1012 I suggest
+
+        Y = self._predict(X=X)
+
+        return Y
+
+    def update(self, X, Y=None):
+        """update model with new data and optional ground truth annotations
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            training data to update model with, time series
+        Y : pd.Series, optional
+            ground truth annotations for training if annotator is supervised
+        Returns
+        -------
+        self : returns a reference to self
+
+        State change
+        ------------
+        updates fitted model (attributes ending in "_")
+        """
+
+        self.check_is_fitted()
+
+        X = check_series(X)
+
+        if Y is not None:
+            Y = check_series(Y)
+
+        self._X = self._X.append(X)
+
+        if Y is not None:
+            self._Y.append(Y)
+
+        self._update(X=X, Y=Y)
+
+        return self
+
+    def update_predict(self, X):
+        """update model with new data and create annotations for it
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            training data to update model with, time series
+
+        Returns
+        -------
+        Y : pd.Series - annotations for sequence X
+            exact format depends on annotation type
+
+        State change
+        ------------
+        updates fitted model (attributes ending in "_")
+        """
+
+        X = check_series(X)
+
+        self.update(X=X)
+        Y = self.predict(X)
+
+        return Y
+
+    def transform(self, Z, Y=None):
```

Comment:
If we want to treat this as a transformer, should it not also inherit from one of the transformer types? It may be easier to implement this as a wrapper.

Reply:
what would we get from inheriting?

Reply:
maybe, but do we get anything from the wrapper that isn't already addressed by this interface alias?

Reply:
this was inconsistent with the `fit`, so I added aliasing (of `X` and `Z`) to all `fit`-like methods

Reply:
I actually removed that agin. I think now it was a bad idea.

We may want to have some aliasing or not, but first we should refactor the transformers modules.
I decided this is out of scope for this PR, probably you agree.